src/napari_imagej/types/widget_mappings.py

```diff
@@ -36,8 +36,13 @@ def preferred_widget_for(
     :param type_hint: The PYTHON type for the parameter
     :return: The best magicgui widget type, if it is known
     """
-    if type_hint == "napari.layers.Image" and item.isInput() and item.isOutput():
-        return "napari_imagej.widgets.parameter_widgets.MutableOutputWidget"
+    if item.isInput() and item.isOutput():
+        if get_origin(type_hint) is Union and type(None) in get_args(type_hint):
+            type_hint = get_args(type_hint)[0]
+            if isinstance(type_hint, ForwardRef):
+                type_hint = type_hint.__forward_arg__
+        if type_hint == "napari.layers.Image":
+            return "napari_imagej.widgets.parameter_widgets.MutableOutputWidget"
```

Comment:
I feel like this is *super* confusing. Can we at least do it within a separate function/

Reply:
Cleaned up in 0bb96a3